pkg/controller/ephemeraljob/ephemeraljob_utils.go

```diff
@@ -0,0 +1,147 @@
+package ephemeraljob
+
+import (
+	"time"
+
+	appsv1alpha1 "github.com/openkruise/kruise/apis/apps/v1alpha1"
+	"github.com/openkruise/kruise/pkg/controller/ephemeraljob/econtainer"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+// pastActiveDeadline checks if job has ActiveDeadlineSeconds field set and if it is exceeded.
+func pastActiveDeadline(job *appsv1alpha1.EphemeralJob) bool {
+	if job.Spec.ActiveDeadlineSeconds == nil || job.Status.StartTime == nil {
+		return false
+	}
+	now := metav1.Now()
+	start := job.Status.StartTime.Time
+	duration := now.Time.Sub(start)
+	allowedDuration := time.Duration(*job.Spec.ActiveDeadlineSeconds) * time.Second
+	return duration >= allowedDuration
+}
+
+func podMatchedEphemeralJob(pod *v1.Pod, ejob *appsv1alpha1.EphemeralJob) (bool, error) {
+	// if selector not matched, then continue
+	if pod.Namespace != ejob.Namespace {
+		return false, nil
+	}
+	selector, err := metav1.LabelSelectorAsSelector(ejob.Spec.Selector)
+	if err != nil {
+		return false, err
+	}
+	if !selector.Empty() && selector.Matches(labels.Set(pod.Labels)) {
+		return true, nil
+	}
+
+	return false, nil
+}
+
+func addConditions(conditions []appsv1alpha1.EphemeralJobCondition, conditionType appsv1alpha1.EphemeralJobConditionType, reason, message string) []appsv1alpha1.EphemeralJobCondition {
+	condition := newCondition(conditionType, reason, message)
+	if len(conditions) == 0 {
+		return []appsv1alpha1.EphemeralJobCondition{
+			condition,
+		}
+	}
+
+	for i := range conditions {
+		if conditions[i].Type == condition.Type {
+			conditions[i] = condition
+			return conditions
+		}
+	}
+
+	conditions = append(conditions, condition)
+	return conditions
+}
+
+func newCondition(conditionType appsv1alpha1.EphemeralJobConditionType, reason, message string) appsv1alpha1.EphemeralJobCondition {
+	return appsv1alpha1.EphemeralJobCondition{
+		Type:               conditionType,
+		Status:             v1.ConditionTrue,
+		LastProbeTime:      metav1.Now(),
+		LastTransitionTime: metav1.Now(),
+		Reason:             reason,
+		Message:            message,
+	}
+}
+
+func timeNow() *metav1.Time {
```

Comment:
timeNow  can be deleted,  we can call metav1.Now() directly without calling this function